frontend/src/scenes/UpgradeModal.tsx

```diff
@@ -11,6 +16,14 @@ export function UpgradeModal(): JSX.Element {
     const { hideUpgradeModal } = useActions(sceneLogic)
 
     const [featureName, featureCaption] = upgradeModalFeatureNameAndCaption ?? []
+    const { featureFlags } = useValues(featureFlagLogic)
+    const { billing } = useValues(billingLogic)
+
+    const { reportSubscriptionStatus } = useActions(eventUsageLogic)
+
+    useEffect(() => {
```

Comment:
I decided against creating a common button component because the language changes also exist on text in the parent components of the buttons. I've added this `useEffect => reportSubscriptionStatus` on all the parents of affected components. BillingHero.tsx, for example won't get this useEffect because its parents OnboardingBillingStep and Billing have the useEffect.